[6.5] UI Port smart variable tests

ldjebran opened the pull request:
Dependency https://github.com/SatelliteQE/airgun/pull/299
Blocked by https://github.com/RedHatQE/widgetastic.core/issues/143

Test results with current airgun workarround, the workarrond is done such a way that we do not need to update tests when the issue is resolved.

```
(robottelo) dlezz@dlezz:~/projects/robottelo-fork$ pytest -v -n 4 tests/foreman/ui_airgun/test_smartvariable.py
============================================ test session starts =============================================
platform linux -- Python 3.6.6, pytest-4.0.2, py-1.5.4, pluggy-0.8.0 -- /home/dlezz/.pyenv/versions/robottelo/bin/python3.6
cachedir: .pytest_cache
shared_function enabled - ON - scope: ffff - storage: redis
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.26.0, timeout-1.3.3, services-1.3.1, mock-1.10.0, forked-0.2, cov-2.6.1
[gw0] linux Python 3.6.6 cwd: /home/dlezz/projects/robottelo-fork
[gw1] linux Python 3.6.6 cwd: /home/dlezz/projects/robottelo-fork
[gw2] linux Python 3.6.6 cwd: /home/dlezz/projects/robottelo-fork
[gw3] linux Python 3.6.6 cwd: /home/dlezz/projects/robottelo-fork
[gw0] Python 3.6.6 (default, Jul 31 2018, 17:47:38)  -- [GCC 5.4.0 20160609]
[gw1] Python 3.6.6 (default, Jul 31 2018, 17:47:38)  -- [GCC 5.4.0 20160609]
[gw2] Python 3.6.6 (default, Jul 31 2018, 17:47:38)  -- [GCC 5.4.0 20160609]
[gw3] Python 3.6.6 (default, Jul 31 2018, 17:47:38)  -- [GCC 5.4.0 20160609]
gw0 [20] / gw1 [20] / gw2 [20] / gw3 [20]
scheduling tests via LoadScheduling

tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_matcher 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_matcher_attribute_priority 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_with_long_priority_list 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_with_host 
[gw3] [  5%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_with_long_priority_list 
tests/foreman/ui_airgun/test_smartvariable.py::test_negative_create_matcher_merge_default 
[gw2] [ 10%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_matcher_attribute_priority 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_matcher_merge_default 
[gw3] [ 15%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_negative_create_matcher_merge_default 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_matcher_avoid_duplicate 
[gw2] [ 20%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_matcher_merge_default 
tests/foreman/ui_airgun/test_smartvariable.py::test_negative_create_matcher_avoid_duplicate 
[gw3] [ 25%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_matcher_avoid_duplicate 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_delete 
[gw2] [ 30%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_negative_create_matcher_avoid_duplicate 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_impact_delete_attribute 
[gw2] [ 35%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_impact_delete_attribute 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_override_default_value_from_attribute 
[gw1] [ 40%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_matcher 
tests/foreman/ui_airgun/test_smartvariable.py::test_negative_create_matcher_merge_override 
[gw0] [ 45%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_with_host 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_matcher_merge_override 
[gw1] [ 50%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_negative_create_matcher_merge_override 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_update_matcher_from_attribute 
[gw0] [ 55%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_matcher_merge_override 
tests/foreman/ui_airgun/test_smartvariable.py::test_negative_update_matcher_from_attribute 
[gw2] [ 60%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_override_default_value_from_attribute 
tests/foreman/ui_airgun/test_smartvariable.py::test_negative_create_override_from_attribute 
[gw0] [ 65%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_negative_update_matcher_from_attribute 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_unhide_default_value_in_attribute 
[gw1] [ 70%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_update_matcher_from_attribute 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_hide_default_value_in_attribute 
[gw2] [ 75%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_negative_create_override_from_attribute 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_update_hidden_value_in_attribute 
[gw3] [ 80%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_delete 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_override_from_attribute 
[gw0] [ 85%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_unhide_default_value_in_attribute 
[gw2] [ 90%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_update_hidden_value_in_attribute 
[gw3] [ 95%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_override_from_attribute 
[gw1] [100%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_hide_default_value_in_attribute 

================================= 20 passed, 60 warnings in 1348.40 seconds ==================================
```

ldjebran commented:
remove do not merge as not more blocked
test outpu to latest changes to dependent airgun PR
```
(robottelo) dlezz@dlezz:~/projects/robottelo-fork$ pytest -v -n 4 tests/foreman/ui_airgun/test_smartvariable.py
============================================ test session starts =============================================
platform linux -- Python 3.6.6, pytest-4.0.2, py-1.5.4, pluggy-0.8.0 -- /home/dlezz/.pyenv/versions/robottelo/bin/python3.6
cachedir: .pytest_cache
shared_function enabled - ON - scope: ffff - storage: redis
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.26.0, timeout-1.3.3, services-1.3.1, mock-1.10.0, forked-0.2, cov-2.6.1
[gw0] linux Python 3.6.6 cwd: /home/dlezz/projects/robottelo-fork
[gw1] linux Python 3.6.6 cwd: /home/dlezz/projects/robottelo-fork
[gw2] linux Python 3.6.6 cwd: /home/dlezz/projects/robottelo-fork
[gw3] linux Python 3.6.6 cwd: /home/dlezz/projects/robottelo-fork
[gw0] Python 3.6.6 (default, Jul 31 2018, 17:47:38)  -- [GCC 5.4.0 20160609]
[gw2] Python 3.6.6 (default, Jul 31 2018, 17:47:38)  -- [GCC 5.4.0 20160609]
[gw3] Python 3.6.6 (default, Jul 31 2018, 17:47:38)  -- [GCC 5.4.0 20160609]
[gw1] Python 3.6.6 (default, Jul 31 2018, 17:47:38)  -- [GCC 5.4.0 20160609]
gw0 [20] / gw1 [20] / gw2 [20] / gw3 [20]
scheduling tests via LoadScheduling

tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_matcher_attribute_priority 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_with_long_priority_list 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_matcher 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_with_host 
[gw1] [  5%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_with_long_priority_list 
tests/foreman/ui_airgun/test_smartvariable.py::test_negative_create_matcher_merge_default 
[gw3] [ 10%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_matcher_attribute_priority 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_matcher_merge_default 
[gw1] [ 15%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_negative_create_matcher_merge_default 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_matcher_avoid_duplicate 
[gw3] [ 20%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_matcher_merge_default 
tests/foreman/ui_airgun/test_smartvariable.py::test_negative_create_matcher_avoid_duplicate 
[gw0] [ 25%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_with_host 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_matcher_merge_override 
[gw2] [ 30%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_matcher 
tests/foreman/ui_airgun/test_smartvariable.py::test_negative_create_matcher_merge_override 
[gw1] [ 35%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_matcher_avoid_duplicate 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_delete 
[gw3] [ 40%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_negative_create_matcher_avoid_duplicate 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_impact_delete_attribute 
[gw0] [ 45%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_matcher_merge_override 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_override_from_attribute 
[gw2] [ 50%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_negative_create_matcher_merge_override 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_override_default_value_from_attribute 
[gw3] [ 55%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_impact_delete_attribute 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_update_matcher_from_attribute 
[gw0] [ 60%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_override_from_attribute 
tests/foreman/ui_airgun/test_smartvariable.py::test_negative_update_matcher_from_attribute 
[gw2] [ 65%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_override_default_value_from_attribute 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_hide_default_value_in_attribute 
[gw3] [ 70%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_update_matcher_from_attribute 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_unhide_default_value_in_attribute 
[gw3] [ 75%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_unhide_default_value_in_attribute 
[gw1] [ 80%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_delete 
tests/foreman/ui_airgun/test_smartvariable.py::test_negative_create_override_from_attribute 
[gw0] [ 85%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_negative_update_matcher_from_attribute 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_update_hidden_value_in_attribute 
[gw2] [ 90%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_hide_default_value_in_attribute 
[gw1] [ 95%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_negative_create_override_from_attribute 
[gw0] [100%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_update_hidden_value_in_attribute 

================================== 20 passed, 60 warnings in 889.56 seconds ==================================
```

ldjebran commented:
test output to latest changes
```
(robottelo) dlezz@dlezz:~/projects/robottelo-fork$ pytest -v -n 4 tests/foreman/ui_airgun/test_smartvariable.py
============================================ test session starts =============================================
platform linux -- Python 3.6.6, pytest-4.0.2, py-1.5.4, pluggy-0.8.0 -- /home/dlezz/.pyenv/versions/robottelo/bin/python3.6
cachedir: .pytest_cache
shared_function enabled - ON - scope: ffff - storage: redis
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.26.0, timeout-1.3.3, services-1.3.1, mock-1.10.0, forked-0.2, cov-2.6.1
[gw0] linux Python 3.6.6 cwd: /home/dlezz/projects/robottelo-fork
[gw1] linux Python 3.6.6 cwd: /home/dlezz/projects/robottelo-fork
[gw2] linux Python 3.6.6 cwd: /home/dlezz/projects/robottelo-fork
[gw3] linux Python 3.6.6 cwd: /home/dlezz/projects/robottelo-fork
[gw0] Python 3.6.6 (default, Jul 31 2018, 17:47:38)  -- [GCC 5.4.0 20160609]
[gw2] Python 3.6.6 (default, Jul 31 2018, 17:47:38)  -- [GCC 5.4.0 20160609]
[gw1] Python 3.6.6 (default, Jul 31 2018, 17:47:38)  -- [GCC 5.4.0 20160609]
[gw3] Python 3.6.6 (default, Jul 31 2018, 17:47:38)  -- [GCC 5.4.0 20160609]
gw0 [20] / gw1 [20] / gw2 [20] / gw3 [20]
scheduling tests via LoadScheduling

tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_matcher_attribute_priority 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_matcher 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_with_host 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_with_long_priority_list 
[gw3] [  5%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_with_long_priority_list 
tests/foreman/ui_airgun/test_smartvariable.py::test_negative_create_matcher_merge_default 
[gw1] [ 10%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_matcher_attribute_priority 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_matcher_merge_default 
[gw3] [ 15%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_negative_create_matcher_merge_default 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_matcher_avoid_duplicate 
[gw1] [ 20%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_matcher_merge_default 
tests/foreman/ui_airgun/test_smartvariable.py::test_negative_create_matcher_avoid_duplicate 
[gw0] [ 25%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_with_host 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_matcher_merge_override 
[gw2] [ 30%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_matcher 
tests/foreman/ui_airgun/test_smartvariable.py::test_negative_create_matcher_merge_override 
[gw3] [ 35%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_matcher_avoid_duplicate 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_delete 
[gw1] [ 40%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_negative_create_matcher_avoid_duplicate 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_impact_delete_attribute 
[gw2] [ 45%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_negative_create_matcher_merge_override 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_override_default_value_from_attribute 
[gw0] [ 50%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_matcher_merge_override 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_override_from_attribute 
[gw1] [ 55%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_impact_delete_attribute 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_update_matcher_from_attribute 
[gw2] [ 60%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_override_default_value_from_attribute 
tests/foreman/ui_airgun/test_smartvariable.py::test_negative_update_matcher_from_attribute 
[gw0] [ 65%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_override_from_attribute 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_hide_default_value_in_attribute 
[gw1] [ 70%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_update_matcher_from_attribute 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_unhide_default_value_in_attribute 
[gw3] [ 75%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_delete 
tests/foreman/ui_airgun/test_smartvariable.py::test_negative_create_override_from_attribute 
[gw0] [ 80%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_hide_default_value_in_attribute 
[gw2] [ 85%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_negative_update_matcher_from_attribute 
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_update_hidden_value_in_attribute 
[gw1] [ 90%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_unhide_default_value_in_attribute 
[gw3] [ 95%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_negative_create_override_from_attribute 
[gw2] [100%] PASSED tests/foreman/ui_airgun/test_smartvariable.py::test_positive_update_hidden_value_in_attribute 

================================== 20 passed, 60 warnings in 870.81 seconds ==================================
2019-04-11 14:40:12 - conftest - DEBUG - Registering custom pytest_namespace

```

oshtaier commented on tests/foreman/ui_airgun/test_smartvariable.py:
```diff
@@ -248,3 +248,948 @@ def test_positive_create_matcher_attribute_priority(
+
+    :expectedresults:
+
+        1.  Error thrown for invalid type value.
```
I see no validation for that. Do we have pop-up there? Could you add proper assertion?

ldjebran replied:
addressed

oshtaier commented on tests/foreman/ui_airgun/test_smartvariable.py:
```diff
@@ -248,3 +248,948 @@ def test_positive_create_matcher_attribute_priority(
+
+    :expectedresults:
+
+        1.  Error thrown for invalid value.
```
same here. We need to check that proper error is shown on UI

ldjebran replied:
addressed

oshtaier commented on tests/foreman/ui_airgun/test_smartvariable.py:
```diff
@@ -248,3 +248,948 @@ def test_positive_create_matcher_attribute_priority(
+            'variable.puppet_class': puppet_class.name,
+            'variable.default_value': variable_default_value,
+            'variable.hidden': True,
+
```
redundant empty line

oshtaier commented on tests/foreman/ui_airgun/test_smartvariable.py:
```diff
@@ -248,3 +248,948 @@ def test_positive_create_matcher_attribute_priority(
+
+        1.  The YAML output has the values merged from all the associated
+            matchers.
+        2.  The YAML output has the default value of variable.
```
`5.  Select 'Merge default' checkbox.` was forgotten, so yeah, it should be:
```
            'variable.prioritize_attribute_order.merge_overrides': True,
            'variable.prioritize_attribute_order.merge_default': True,
            'variable.prioritize_attribute_order.avoid_duplicates': True,
```
and
`assert output['parameters'][name] == [20, 80, 90, 100]`

oshtaier replied:
very nice catch @mirzal

ldjebran replied:
addressed thanks @mirzal and @oshtaier

ldjebran commented on tests/foreman/ui_airgun/test_smartvariable.py:
```diff
@@ -248,3 +248,948 @@ def test_positive_create_matcher_attribute_priority(
+        3.  Create second matcher for other attribute with valid details.
+            Note - The fqdn/host should have this attribute.
+        4.  Create more matchers for some more attributes if any.
+            Note - The fqdn/host should have this attributes.
```
Yes this step seems not mandatory as it say "if any" propose to remove this step completely @mirzal @oshtaier  what do you thing?

mirzal replied:
Yeah, I would remove it. It doesn't describe what automation is doing and nobody is going to read it (since case is automated).

oshtaier replied:
yeap, automating that step will increase scope dramatically, just remove it

ldjebran replied:
addressed

ldjebran commented:
@mirzal @oshtaier addressed all comments
```
(robottelo) dlezz@dlezz:~/projects/robottelo-fork$ pytest -v tests/foreman/ui_airgun/test_smartvariable.py
============================================ test session starts =============================================
platform linux -- Python 3.6.6, pytest-4.0.2, py-1.5.4, pluggy-0.8.0 -- /home/dlezz/.pyenv/versions/robottelo/bin/python3.6
cachedir: .pytest_cache
shared_function enabled - ON - scope: ffff - storage: redis
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.26.0, timeout-1.3.3, services-1.3.1, mock-1.10.0, forked-0.2, cov-2.6.1
collecting ... 2019-04-12 16:27:42 - conftest - DEBUG - BZ deselect is disabled in settings

collected 20 items                                                                                           

tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_with_host PASSED                   [  5%]
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_matcher PASSED                     [ 10%]
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_matcher_attribute_priority PASSED  [ 15%]
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_with_long_priority_list PASSED     [ 20%]
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_matcher_merge_override PASSED      [ 25%]
tests/foreman/ui_airgun/test_smartvariable.py::test_negative_create_matcher_merge_override PASSED      [ 30%]
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_matcher_merge_default PASSED       [ 35%]
tests/foreman/ui_airgun/test_smartvariable.py::test_negative_create_matcher_merge_default PASSED       [ 40%]
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_create_matcher_avoid_duplicate PASSED     [ 45%]
tests/foreman/ui_airgun/test_smartvariable.py::test_negative_create_matcher_avoid_duplicate PASSED     [ 50%]
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_delete PASSED                             [ 55%]
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_impact_delete_attribute PASSED            [ 60%]
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_override_from_attribute PASSED            [ 65%]
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_override_default_value_from_attribute PASSED [ 70%]
tests/foreman/ui_airgun/test_smartvariable.py::test_negative_create_override_from_attribute PASSED     [ 75%]
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_update_matcher_from_attribute PASSED      [ 80%]
tests/foreman/ui_airgun/test_smartvariable.py::test_negative_update_matcher_from_attribute PASSED      [ 85%]
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_hide_default_value_in_attribute PASSED    [ 90%]
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_unhide_default_value_in_attribute PASSED  [ 95%]
tests/foreman/ui_airgun/test_smartvariable.py::test_positive_update_hidden_value_in_attribute PASSED   [100%]

================================= 20 passed, 48 warnings in 2459.42 seconds ==================================
```
